# Baik Baik
## A Dataviz About Bikes and Daily Life in Paris

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec in odio sem. Cras venenatis quam nec hendrerit laoreet. Suspendisse in odio et elit commodo aliquet. Sed libero turpis, rhoncus at finibus eu, vehicula ut est. Sed sem mi, aliquam id rhoncus quis, semper in nisi. Aliquam erat volutpat. Vestibulum auctor risus at ipsum pretium mollis. Donec non sapien et velit elementum placerat ut in orci.

Pellentesque iaculis sagittis ligula. Donec aliquam sapien ac fringilla iaculis. Pellentesque imperdiet facilisis massa. Phasellus in enim eu mauris ornare egestas nec a augue. Duis sed tincidunt arcu, vitae cursus nisl. Ut at elit neque. Nunc tellus urna, consectetur non lectus at, ultricies maximus turpis. Proin sit amet odio vitae est luctus tincidunt. Curabitur semper enim nisi, ut aliquet sem condimentum non. 

### Context

Passion for velib --> got the data

### Getting into it

#### The Data

Let's see what this .csv file has into it.

First, we have to install some useful libraries which are gonna help us later.

In [1]:
# For exploring and working with data
import numpy as np
import pandas as pd

# For coping with date formats
import datetime

# For plotting and visualizing
import matplotlib as plt 
import seaborn as sns

Now we can read our CSV files and explore the data to give us some insights about what we can do.

In [7]:
data_velib = pd.read_csv("data/velib_data.csv")

# Returns a tuple as (number_of_lines, number_of_columns)
data_velib.shape

(412, 20)

In [8]:
# Let's print the first five rows of the dataset
data_velib.head()

ID location     ID abonné  ID Abonnement Type d'abonnement  ID Vélo  \
0  1,806,565,011  9.425151e+16         821238            V-Plus  21329.0   
1  1,807,295,310  9.425151e+16         821238            V-Plus    195.0   
2  1,843,776,641  9.425151e+16         821238            V-Plus  24339.0   
3  1,847,609,163  9.425151e+16         821238            V-Plus   7840.0   
4  1,851,069,210  9.425151e+16         821238            V-Plus  22372.0   

  Type de vélo  Version vBox Date/heure de début de la location  \
0           VM          3.38                    7/24/2019 11:55   
1           VM          3.38                    7/24/2019 14:14   
2           VM          3.38                    7/28/2019 14:20   
3           VM          3.38                     7/29/2019 7:31   
4           VM          3.38                    7/29/2019 16:28   

   ID station départ                   Nom station départ  CP station départ  \
0              10025      Jemmapes - Ecluses Saint-Martin              75010   
1               4104               Sebastopol - Rambuteau              75004   
2              10151  Gare du Nord - Faubourg Saint-Denis              75010   
3              17108          Bessières - Porte de Clichy              75017   
4              17108          Bessières - Porte de Clichy              75017   

  Ville départ Date/heure de fin de la location  ID station arrivée  \
0     PARIS 10                  7/24/2019 12:11                4104   
1     PARIS 04                  7/24/2019 14:32               10025   
2     PARIS 10                  7/28/2019 14:23               10026   
3     PARIS 17                   7/29/2019 7:41               17023   
4     PARIS 17                  7/29/2019 16:50               10162   

                Nom station arrivée  CP station arrivée Ville arrivée  \
0            Sebastopol - Rambuteau               75004      PARIS 04   
1   Jemmapes - Ecluses Saint-Martin               75010      PARIS 10   
2    Gare de l'Est - Chateau Landon               75010      PARIS 10   
3                  Porte d'Asnières               75017      PARIS 17   
4  Alexandre Parodi - Quai de Valmy               75010      PARIS 10   

   Durée location (mn)  Distance parcourue en mètres  \
0                16.48                          2721   
1                18.30                          3087   
2                 2.63                           553   
3                10.11                          2131   
4                22.43                          5787   

   Montant de la course en euros  
0                              0  
1                              0  
2                              0  
3                              0  
4                              0

Wow, that's a lot of information ! 

We have 412 trips recorded in the dataset, that's quite enough for us to start our study. <br>
There is also 20 columns describing in details every trip from the bike used to date and location from departure and arrival as well as the distance and the location duration.

Some of the data is likely to be poorly relevant for our analysis (location ID or vBox version for example).


#### Research Questions

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec in odio sem. Cras venenatis quam nec hendrerit laoreet. Suspendisse in odio et elit commodo aliquet. Sed libero turpis, rhoncus at finibus eu, vehicula ut est. Sed sem mi, aliquam id rhoncus quis, semper in nisi. Aliquam erat volutpat. Vestibulum auctor risus at ipsum pretium mollis. Donec non sapien et velit elementum placerat ut in orci.

Pellentesque iaculis sagittis ligula. Donec aliquam sapien ac fringilla iaculis. Pellentesque imperdiet facilisis massa. Phasellus in enim eu mauris ornare egestas nec a augue. Duis sed tincidunt arcu, vitae cursus nisl. Ut at elit neque. Nunc tellus urna, consectetur non lectus at, ultricies maximus turpis. Proin sit amet odio vitae est luctus tincidunt. Curabitur semper enim nisi, ut aliquet sem condimentum non. 

#### How the Dataset Can Help Us

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec in odio sem. Cras venenatis quam nec hendrerit laoreet. Suspendisse in odio et elit commodo aliquet. Sed libero turpis, rhoncus at finibus eu, vehicula ut est. Sed sem mi, aliquam id rhoncus quis, semper in nisi. Aliquam erat volutpat. Vestibulum auctor risus at ipsum pretium mollis. Donec non sapien et velit elementum placerat ut in orci.

Pellentesque iaculis sagittis ligula. Donec aliquam sapien ac fringilla iaculis. Pellentesque imperdiet facilisis massa. Phasellus in enim eu mauris ornare egestas nec a augue. Duis sed tincidunt arcu, vitae cursus nisl. Ut at elit neque. Nunc tellus urna, consectetur non lectus at, ultricies maximus turpis. Proin sit amet odio vitae est luctus tincidunt. Curabitur semper enim nisi, ut aliquet sem condimentum non. 

#### Cleaning the Data

Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec in odio sem. Cras venenatis quam nec hendrerit laoreet. Suspendisse in odio et elit commodo aliquet. Sed libero turpis, rhoncus at finibus eu, vehicula ut est. Sed sem mi, aliquam id rhoncus quis, semper in nisi. Aliquam erat volutpat. Vestibulum auctor risus at ipsum pretium mollis. Donec non sapien et velit elementum placerat ut in orci.

Pellentesque iaculis sagittis ligula. Donec aliquam sapien ac fringilla iaculis. Pellentesque imperdiet facilisis massa. Phasellus in enim eu mauris ornare egestas nec a augue. Duis sed tincidunt arcu, vitae cursus nisl. Ut at elit neque. Nunc tellus urna, consectetur non lectus at, ultricies maximus turpis. Proin sit amet odio vitae est luctus tincidunt. Curabitur semper enim nisi, ut aliquet sem condimentum non. 

### Diving into it